From 05e25e9521d4629ee8d24e0c45c49831df3f1e50 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Tue, 13 Nov 2018 22:44:04 +0100
Subject: [PATCH 4/4] pie-gsi tracking

commit 2cb443950e8203d3c63b48a97a76af847432f94d
Merge: 1df26dbc4 3058e6167
Author: Treehugger Robot <treehugger-gerrit@google.com>
Date:   Mon Oct 22 07:40:30 2018 +0000

    Merge "Workaround Legancy GSI for some Wifi firmware" into pie-gsi

commit 3058e61672903f386f63544e02e1cc9cd17cf8d5
Author: SzuWei Lin <szuweilin@google.com>
Date:   Fri Oct 19 16:40:33 2018 +0800

    Workaround Legancy GSI for some Wifi firmware

    Bug: 117963911
    Test: Turn on wifi with aosp_arm64_ab-userdebug on Pixel 1/2/3 devices
    Change-Id: I0d516876522a687709ff21cbfc5bc0cc415b8335
    (cherry picked from commit 11892a4b9aa50a79803082ba2aa3bdf54b925681)

commit 1df26dbc4876bdb9b08d6c3cddc45551fa9b406b
Merge: 6ae5841ae b49e8e5d0
Author: Android Merger <noreply-android-build-merger@google.com>
Date:   Wed Oct 17 04:22:42 2018 +0000

    Merge "Build prefix on pi-platform-release is PPRL. am: cd2fda4e68" into pie-gsi

commit b49e8e5d0303a3e95197a70af53f6cd793b4cc1e
Merge: 6ae5841ae cd2fda4e6
Author: Bill Rassieur <rassb@google.com>
Date:   Tue Oct 16 21:22:37 2018 -0700

    Build prefix on pi-platform-release is PPRL.
    am: cd2fda4e68

    Change-Id: I192bad1d4a0113629216cbd8adf63051650b308e

commit 6ae5841ae47a583544ce64ea19d5c7482834beca
Merge: 5b7a4ed39 cbe03c23e
Author: Android Merger <noreply-android-build-merger@google.com>
Date:   Wed Oct 17 04:17:47 2018 +0000

    Merge "Merge PPR2.181005.003 from pi-release-2 into pi-platform-release. am: 6d4dec0f94" into pie-gsi

commit cbe03c23e9ac4724a52c7b78399593ce632b8211
Merge: 5b7a4ed39 6d4dec0f9
Author: Bill Rassieur <rassb@google.com>
Date:   Tue Oct 16 21:17:41 2018 -0700

    Merge PPR2.181005.003 from pi-release-2 into pi-platform-release.
    am: 6d4dec0f94

    Change-Id: I4d4bac8775480fff8b9be87b78cd2ffecf3b51d2

commit 5b7a4ed39f0fe4ae76282b9022643406599867e2
Merge: 05359c42c 37e218e53
Author: Treehugger Robot <treehugger-gerrit@google.com>
Date:   Mon Oct 15 07:11:58 2018 +0000

    Merge "Add build target aosp_arm_64b_a" into pie-gsi

commit 37e218e53b43cd536205bc9c39fc1552d159cb4e
Author: SzuWei Lin <szuweilin@google.com>
Date:   Mon Oct 15 11:59:37 2018 +0800

    Add build target aosp_arm_64b_a

    The GSI build target is for legacy devices which upgrded from
    O/O-MR1. The configure is:

    - ARM 32bits user space
    - 64bits binder interface
    - non-system-as-root partition layout

    Bug: 117576577
    Test: lunch aosp_arm_64b_a-userdebug; make -j
    Change-Id: Ia51e154863cbf0a103d7d4fc96d08db883625568

commit cd2fda4e689df5ba7672a1b92f4cea17be3ede26
Author: Bill Rassieur <rassb@google.com>
Date:   Thu Oct 11 09:28:15 2018 -0700

    Build prefix on pi-platform-release is PPRL.

    BUG: 117431430

commit 6d4dec0f94851fbde5c3d246824eda9b7a70ee51
Merge: f9d90a678 fcae27240
Author: Bill Rassieur <rassb@google.com>
Date:   Wed Oct 10 13:14:07 2018 -0700

    Merge PPR2.181005.003 from pi-release-2 into pi-platform-release.

    BUG: 117431430

Change-Id: I535ddd807d95cf128776185170cdee4241676270
---
 target/board/generic_arm64/BoardConfig.mk     |  2 +-
 target/board/generic_arm64_ab/BoardConfig.mk  |  4 ++-
 target/board/generic_arm_64b_a/BoardConfig.mk | 31 +++++++++++++++++++
 target/board/generic_arm_ab/BoardConfig.mk    |  4 ++-
 target/product/AndroidProducts.mk             |  1 +
 target/product/aosp_arm_64b_a.mk              | 27 ++++++++++++++++
 6 files changed, 66 insertions(+), 3 deletions(-)
 create mode 100644 target/board/generic_arm_64b_a/BoardConfig.mk
 create mode 100644 target/product/aosp_arm_64b_a.mk

diff --git a/target/board/generic_arm64/BoardConfig.mk b/target/board/generic_arm64/BoardConfig.mk
index c1983ad46..13d5bd64a 100644
--- a/target/board/generic_arm64/BoardConfig.mk
+++ b/target/board/generic_arm64/BoardConfig.mk
@@ -61,7 +61,7 @@ BOARD_USERDATAIMAGE_PARTITION_SIZE := 576716800
 # cleaned up all device specific directories under root!
 
 # TODO(b/111434759, b/111287060) SoC specific hacks
-BOARD_ROOT_EXTRA_SYMLINKS := /vendor/lib/dsp:/dsp
+BOARD_ROOT_EXTRA_SYMLINKS += /vendor/lib/dsp:/dsp
 BOARD_ROOT_EXTRA_SYMLINKS += /mnt/vendor/persist:/persist
 BOARD_ROOT_EXTRA_SYMLINKS += /vendor/firmware_mnt:/firmware
 
diff --git a/target/board/generic_arm64_ab/BoardConfig.mk b/target/board/generic_arm64_ab/BoardConfig.mk
index fc6b58284..88b90a8fb 100644
--- a/target/board/generic_arm64_ab/BoardConfig.mk
+++ b/target/board/generic_arm64_ab/BoardConfig.mk
@@ -34,7 +34,9 @@ BOARD_BUILD_SYSTEM_ROOT_IMAGE := true
 
 # TODO(jiyong) These might be SoC specific.
 BOARD_ROOT_EXTRA_FOLDERS += firmware firmware/radio persist
-BOARD_ROOT_EXTRA_SYMLINKS := /vendor/lib/dsp:/dsp
+BOARD_ROOT_EXTRA_SYMLINKS += /vendor/lib/dsp:/dsp
+BOARD_ROOT_EXTRA_SYMLINKS += /vendor/firmware_mnt/image:/firmware/image
+BOARD_ROOT_EXTRA_SYMLINKS += /vendor/firmware_mnt/verinfo:/firmware/verinfo
 
 # Set this to create /cache mount point for non-A/B devices that mounts /cache.
 # The partition size doesn't matter, just to make build pass.
diff --git a/target/board/generic_arm_64b_a/BoardConfig.mk b/target/board/generic_arm_64b_a/BoardConfig.mk
new file mode 100644
index 000000000..fa9cfe71a
--- /dev/null
+++ b/target/board/generic_arm_64b_a/BoardConfig.mk
@@ -0,0 +1,31 @@
+#
+# Copyright (C) 2018 The Android Open-Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+include build/make/target/board/treble_common_32.mk
+
+# For devices using 64-bit binder interface
+TARGET_USES_64_BIT_BINDER := true
+
+TARGET_ARCH := arm
+TARGET_ARCH_VARIANT := armv7-a-neon
+TARGET_CPU_ABI := armeabi-v7a
+TARGET_CPU_ABI2 := armeabi
+TARGET_CPU_VARIANT := generic
+
+# Set this to create /cache mount point for non-A/B devices that mounts /cache.
+# The partition size doesn't matter, just to make build pass.
+BOARD_CACHEIMAGE_FILE_SYSTEM_TYPE := ext4
+BOARD_CACHEIMAGE_PARTITION_SIZE := 16777216
diff --git a/target/board/generic_arm_ab/BoardConfig.mk b/target/board/generic_arm_ab/BoardConfig.mk
index 7d9ea9c0e..3d1484290 100644
--- a/target/board/generic_arm_ab/BoardConfig.mk
+++ b/target/board/generic_arm_ab/BoardConfig.mk
@@ -28,7 +28,9 @@ BOARD_BUILD_SYSTEM_ROOT_IMAGE := true
 
 # TODO(jiyong) These might be SoC specific.
 BOARD_ROOT_EXTRA_FOLDERS += firmware firmware/radio persist
-BOARD_ROOT_EXTRA_SYMLINKS := /vendor/lib/dsp:/dsp
+BOARD_ROOT_EXTRA_SYMLINKS += /vendor/lib/dsp:/dsp
+BOARD_ROOT_EXTRA_SYMLINKS += /vendor/firmware_mnt/image:/firmware/image
+BOARD_ROOT_EXTRA_SYMLINKS += /vendor/firmware_mnt/verinfo:/firmware/verinfo
 
 # Set this to create /cache mount point for non-A/B devices that mounts /cache.
 # The partition size doesn't matter, just to make build pass.
diff --git a/target/product/AndroidProducts.mk b/target/product/AndroidProducts.mk
index fa2ad12c8..7237ea689 100644
--- a/target/product/AndroidProducts.mk
+++ b/target/product/AndroidProducts.mk
@@ -52,6 +52,7 @@ PRODUCT_MAKEFILES := \
     $(LOCAL_DIR)/aosp_arm.mk \
     $(LOCAL_DIR)/aosp_arm_a.mk \
     $(LOCAL_DIR)/aosp_arm_ab.mk \
+    $(LOCAL_DIR)/aosp_arm_64b_a.mk \
     $(LOCAL_DIR)/aosp_arm_64b_ab.mk \
     $(LOCAL_DIR)/full.mk \
     $(LOCAL_DIR)/aosp_x86.mk \
diff --git a/target/product/aosp_arm_64b_a.mk b/target/product/aosp_arm_64b_a.mk
new file mode 100644
index 000000000..18e02b86a
--- /dev/null
+++ b/target/product/aosp_arm_64b_a.mk
@@ -0,0 +1,27 @@
+#
+# Copyright (C) 2018 The Android Open-Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# PRODUCT_PROPERTY_OVERRIDES cannot be used here because sysprops will be at
+# /vendor/[build|default].prop when build split is on. In order to have sysprops
+# on the generic system image, place them in build/make/target/board/
+# treble_system.prop.
+
+include build/make/target/product/treble_common_32.mk
+
+PRODUCT_NAME := aosp_arm_64b_a
+PRODUCT_DEVICE := generic_arm_64b_a
+PRODUCT_BRAND := Android
+PRODUCT_MODEL := AOSP on ARM32
-- 
2.17.1

